libs/wire-api/src/Wire/API/User.hs

```diff
@@ -492,6 +494,19 @@ validateNewUserPublic nu
   | otherwise =
     Right (NewUserPublic nu)
 
+-- | Any user registering without either an email or a phone is Ephemeral,
+-- i.e. can be deleted after expires_in or sessionTokenTimeout
+isNewUserEphemeral :: NewUser -> Bool
+isNewUserEphemeral u = case newUserIdentity u of
+  Nothing -> True
+  Just _ -> False
+
+isNewUserTeamMember :: NewUser -> Bool
+isNewUserTeamMember u = case newUserTeam u of
+  Just (NewTeamMember _) -> True
+  Just (NewTeamMemberSSO _) -> True
+  _ -> False
```

Comment:
Opinion poll: should we be explicit about all cases to avoid having false assumptions in case we add more constructors to `NewTeamUser`?

Reply:
In my head we already have that rule, but I suppose we should write it down somewhere...  :)